lib/index.js

```diff
@@ -57,32 +57,85 @@ module.exports = (function() {
         return obj === undefined ? defaultValue : obj;
     }
 
-   /**
+		/**
     * Delete unused sheets if needed
     */
     Workbook.prototype.deleteSheet = function(sheetName){
       var self = this;
       var sheet = self.loadSheet(sheetName);
 
-      Object.keys(self.archive.files).forEach(function(key){
-        if (key.indexOf(sheet.id) != -1)
-        delete self.archive.files[key];
-      })
+      var sh = self.workbook.find("sheets/sheet[@sheetId='" + sheet.id + "']");
+      self.workbook.find("sheets").remove(sh);
+			
+      var rel = self.workbookRels.find("Relationship[@Id='" + sh.attrib['r:id'] + "']");
+      self.workbookRels.remove(rel);
+
+      self._rebuild();
+      return self
+    };
 
-      var id = self.sheets.findIndex(function(el){return el.id == sheet.id});
-      self.sheets.splice(id, 1);
+    /**
+    * Clone sheets in current workbook template
+    */
+    Workbook.prototype.copySheet = function(sheetName, copyName){
+      var self = this;
+      var sheet = self.loadSheet(sheetName); //filename, name , id, root
+      var indexS = (self.workbook.findall("sheets/sheet").length+1).toString();
```

Comment:
Would be good to rename variable to `nextSheetIndex`, or `newSheetIndex`.

Reply:
Done